Gifski/EditVideoViewController.swift

```diff
@@ -338,18 +342,33 @@ final class EditVideoViewController: NSViewController {
 			}
 		}()
 		let frameCount = duration * frameRateSlider.doubleValue
-		let dimensions = resizableDimensions.changed(dimensionsType: .pixels).currentDimensions.value
-		var fileSize = (Double(dimensions.width) * Double(dimensions.height) * frameCount) / 3
-		fileSize = fileSize * (qualitySlider.doubleValue + 1.5) / 2.5
+
 		estimatedSizeLabel.textColor = NSColor(red: 0.13, green: 0.13, blue: 0.13, alpha: 1.0)
-		estimatedSizeLabel.stringValue = "Estimated size: " + formatter.string(fromByteCount: Int64(fileSize))
+		estimatedSizeLabel.stringValue = "Starting Estimation..."
 		estimateWorkItem = DispatchWorkItem {
+			print("Total: \(frameCount)")
+
+			Gifski.setOnNewFrame { framesWritten in
+				DispatchQueue.main.sync {
+					self.estimatedSizeLabel.textColor = NSColor(red: 0.11, green: 0.37, blue: 0.13, alpha: 1.0)
+				}
+
+				if self.dataSize == 0 {
+					return
+				}
+
+				self.framesWritten = framesWritten
+				DispatchQueue.main.sync {
+					self.estimatedSizeLabel.stringValue = "Estimated size: " + self.formatter.string(fromByteCount: Int64(self.dataSize * (Int64(frameCount) / Int64(framesWritten))))
+				}
+			}
+			Gifski.setWriteCallback { size in
+				self.dataSize += Int64(size)
```

Comment:
the library supports only 1 callback at a time, so this will *replace* the other callback that gets the data.

Reply:
Yes. I am also only converting on thing at a time. Either estimating or actually converting.

Reply:
I thought you also build gifData at the same time?

Reply:
No I set gifData at the end of the conversion.

Reply:
Anything else?

Reply:
Just empty gifData? gifData is appended to from a write callback.

Reply:
the setWriteCallback does not actually override the writeCallback with Gifski it is called after everything else is run onWriteCallback.

Reply:
Gifski.swift still writes the gifData and then I get it to save later on at the end of the conversion

Reply:
Ah, I assumed the setter was just forwarding it to the underlying library instead of adding its own layer of callback handling.